worker/uniter/relation/relationer.go

```diff
@@ -53,15 +53,15 @@ func (r *Relationer) RelationUnit() *apiuniter.RelationUnit {
 
 // Join initializes local state and causes the unit to enter its relation
 // scope, allowing its counterpart units to detect its presence and settings
-// changes. Local state directory is not created until needed.
+// changes.
 func (r *Relationer) Join() error {
 	if r.dying {
 		panic("dying relationer must not join!")
 	}
-	// We need to make sure the state directory exists before we join the
-	// relation, lest a subsequent ReadAllStateDirs report local state that
+	// We need to make sure the state is persisted inState before we join
+	// the relation, lest a subsequent ReadAllStates report local state that
 	// doesn't include relations recorded in remote state.
-	if err := r.dir.Ensure(); err != nil {
+	if err := r.ops.EnsureState(); err != nil {
```

Comment:
Took my a while to figure out what we mean by ensure; since we are refactoring could we rename it to something like `RefreshState` or `ReloadState` that will make sense to us when we revisit the code a few months down the road?